src/containers/Auth/ErrorPage.js

```diff
@@ -25,25 +25,32 @@ const ErrorPage = (props) => {
   const history = useHistory();
   const [errorInfo, setErrorInfo] = useState(null);
   const search = useLocation().search;
-  const errorCodeInput = new URLSearchParams(search).get("error");
+  const errorCodeInput = new URLSearchParams(search).get("id");
 
-  const moveToDashboard = (e) => {
+  const moveToDashboard = () => {
     history.push({
       pathname: "/",
     });
   };
 
-  const getKratosError = (errorCode) =>
+  const getKratosError = (errorCode) => {
     newKratosSdk()
       .getSelfServiceError(errorCode)
-      .then(({ resp }) => {
-        if (resp && resp.length) {
-          setErrorInfo(resp[0]);
+      .then(({ data: resp }) => {
+        if (resp) {
+          setErrorInfo(resp.error);
         }
       })
-      .catch((errorResp) => {
+      .catch((err) => {
         setErrorInfo(null);
+        switch (err.response?.status) {
+          case (404, 403, 410):
```

Comment:
if this line of code executes means it failed to fetch kratos error for some reason but it did reach this point due to an application failure ! Do you think we should just redirect back to dashboard automatically or display below error occurred screen and let user click on "go to dashboard" ?

Reply:
If error flow return 404, 403 or 410, mean the error id not found, some CSRF issue or expired respectively. In this case it is safe to return back user to home page. Same logic has been used in [Ory example here](https://github.com/ory/kratos-selfservice-ui-react-nextjs/blob/161a33fefbeecb6f21fd73da4a91fc21131939b6/pages/error.tsx#L32-L39).

Reply:
thanks @akshay196 for the detailed explaination !